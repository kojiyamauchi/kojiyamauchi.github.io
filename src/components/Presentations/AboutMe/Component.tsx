/*
  About Me Component.
*/

import styles from './style.module.scss'

type Props = {
  closeAboutMe: () => void
}

export const AboutMe: React.FC<Props> = ({ closeAboutMe }): JSX.Element => {
  return (
    <div className={`${styles.wrapper} about-me`}>
      <h1 className={`${styles['primary-heading']} primary-heading`}>
        <span className="is-en">Koji Yamauchi</span>&nbsp;&nbsp;|&nbsp;&nbsp;<span className="is-jp">ヤマウチ コウジ</span>
      </h1>
      <p className={`${styles.letter} letter is-en`}>
        I am mainly involved in web front-end development and production from Based in Tokyo, Japan.
        <br />
        (Some back-end development and design work)
        <br />
        In the front-end area, we can handle everything from styling, such as UI implementation and design implementation, to interaction implementation,
        animation implementation, and API communication, so please feel free to contact us.
        <br />
        My skills, TypeScript / JavaScript / React / Next.js / Node.js / Webpack / Gulp / GitHub / Sass / CSS / HTML / VSCode / Figma(a little) / Sketch(a
        little) / Photoshop / Illustrator.
        <br />
        Currently, past projects is private.
        <br />
        If want you see please following contact us.
        <br />
        I’ll get back to you.
        <br />
        Loves &amp; fun making things.<span className="is-emoji">✌️</span>
      </p>
      <p className={`${styles.letter} letter is-jp`}>
        東京都内にて、主にWebフロントエンドの開発・制作に携わっております。
        <br />
        (Node.js開発・デザイン制作も少々)
        <br />
        フロントエンド領域においては、UI実装・デザイン実装等のスタイリング周りから、インタラクション実装・アニメーション実装・API通信まで承る事が可能ですので、お気軽にご連絡下さい。
        <br />
        使用言語・開発環境等は、TypeScript・JavaScript・React・Next.js・Node.js・Webpack・Gulp・GitHub・Sass・CSS・HTML・VSCode・Figma(少々)・Sketch(少々)・Photoshop・Illustratorになります。
        <br />
        現在、制作実績は非公開になりますので、下記よりお問い合わせ下さい。
        <br />
        折り返しご連絡させていただきます。
      </p>
      <div className={`${styles['mail-wrapper']} mail-wrapper`}>
        <a className={`${styles['button-mail']} button-mail`} href="mailto:all.the.small.things@me.com">
          <svg className={`${styles['icon-arrow-mail']} icon-arrow-mail`} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path d="M4.01 6.03l7.51 3.22-7.52-1 .01-2.22m7.5 8.72L4 17.97v-2.22l7.51-1M2.01 3L2 10l15 2-15 2 .01 7L23 12 2.01 3z" />
          </svg>
          CONTACT US
        </a>
      </div>
      <div className={`${styles['back-wrapper']} back-wrapper`}>
        <button className={`${styles['button-back']} button-back`} onClick={closeAboutMe}>
          <svg className={`${styles['icon-arrow-back']} icon-arrow-back`} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z" />
          </svg>
          HIDE ABOUT ME
        </button>
      </div>
    </div>
  )
}
