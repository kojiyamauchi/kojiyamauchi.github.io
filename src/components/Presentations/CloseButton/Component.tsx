/*
  Close Button Component.
*/

import { useCloseButton } from './Hooks'
import styles from './style.module.scss'

type Props = {
  modalState: boolean
  modalClose: () => void
}

export const CloseButton: React.FC<Props> = ({ modalState, modalClose }): JSX.Element => {
  const { myComponent, letter } = useCloseButton(modalState)

  return (
    <button className={`${styles.wrapper} close-button`} ref={myComponent} onClick={(): void => modalClose()}>
      <span className="close-letters-wrapper">
        {Array.from(letter, (info, index) => (
          <span className={`${styles['close-letter']} close-letter`} key={index}>
            {info}
          </span>
        ))}
      </span>
    </button>
  )
}
