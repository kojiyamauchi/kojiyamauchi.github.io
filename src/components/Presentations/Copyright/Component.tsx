/*
  Copyright Component.
*/

import { useEffect, useState } from 'react'

import styles from './style.module.scss'

type Props = {
  className?: string
  parent: 'footer' | 'modal' | 'about-me' | 'error'
}

export const Copyright: React.VFC<Props> = ({ className, parent }): JSX.Element => {
  const [year, setYear] = useState<number | null>(null)

  useEffect(() => {
    setYear(new Date().getFullYear())
  }, [])

  return (
    <a className={`${styles.wrapper} ${className} is-${parent}`} href="http://kojiyamauchi.com" target="_blank" rel="noreferrer">
      Â© {year} All The Small Things, All Rights Reserved.
    </a>
  )
}
