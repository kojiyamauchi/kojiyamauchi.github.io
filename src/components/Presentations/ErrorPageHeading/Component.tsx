/*
  Error Page Heading Component.
*/
import styles from './style.module.scss'

export const ErrorPageHeading: React.FC = (): JSX.Element => <h1 className={`${styles['primary-heading']} error-page-primary-heading`}>Error Page.</h1>
