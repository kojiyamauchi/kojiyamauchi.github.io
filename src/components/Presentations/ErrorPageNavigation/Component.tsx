/*
  Error Page Navigation Component.
*/

import Link from 'next/link'

import styles from './style.module.scss'

export const ErrorPageNavigation: React.FC = (): JSX.Element => {
  const link = [{ path: '/', letter: 'Back to Top Page', id: 1 }]

  const navigation = (): JSX.Element[] => {
    return link.map(
      (info): JSX.Element => (
        <li key={info.id} className="error-page-navi-list">
          <Link href={`${info.path}`}>
            <a className={`${styles.link} error-page-navi-link`}>{info.letter}</a>
          </Link>
        </li>
      )
    )
  }

  return <ul className={`${styles.wrapper} error-page-navigation`}>{navigation()}</ul>
}
