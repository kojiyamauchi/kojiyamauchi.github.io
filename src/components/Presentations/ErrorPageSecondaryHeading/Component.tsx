/*
  Error Pages Secondary Heading Component.
*/

import styles from './style.module.scss'

type Props = {
  letter: string
  translation?: boolean
}

export const ErrorPageSecondaryHeading: React.FC<Props> = ({ letter, translation }): JSX.Element => (
  <h2 className={`${styles['secondary-heading']} error-page-secondary-heading page-heading ${translation ? styles['is-translation'] : ''}`}>{letter}.ðŸ˜µ</h2>
)
