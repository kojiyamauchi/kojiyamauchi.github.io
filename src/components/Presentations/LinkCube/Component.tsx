/*
  Link Cube Component.
*/

import { DoubleSide } from 'three'

import { useLinkCube } from './Hooks'

export type Props = {
  textureName: string
  linkUrl: string
  positionValue: {
    x: {
      mobile: number
      tablet: number
      deskTop: number
    }
    y: {
      mobile: number
      tablet: number
      deskTop: number
    }
  }
  rotate: {
    x: number
    y: number
    z: number
  }
  storybook?: boolean
}

export const LinkCube: React.FC<Props> = ({ textureName, linkUrl, positionValue, rotate, storybook = false }): JSX.Element => {
  const { myComponent, memorizePositionX, memorizePositionY, memorizeCubeSize, texture } = useLinkCube(textureName, positionValue, rotate, storybook)

  return (
    <mesh
      ref={myComponent}
      position={[memorizePositionX, memorizePositionY, 0]}
      onClick={(): Window | null => open(linkUrl, '_blank')}
      onPointerOver={(): string => (document.body.style.cursor = 'pointer')}
      onPointerOut={(): string => (document.body.style.cursor = 'default')}
    >
      <boxGeometry attach="geometry" args={[memorizeCubeSize, memorizeCubeSize, memorizeCubeSize]} />
      <meshPhongMaterial attach="material" map={texture} side={DoubleSide} color="#ffffff" />
    </mesh>
  )
}
