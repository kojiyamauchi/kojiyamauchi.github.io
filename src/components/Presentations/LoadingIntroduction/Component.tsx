/*
  Loading Introduction Component.
*/

import { useRouter } from 'next/router'

import { useLoadingIntroduction } from './Hooks'
import styles from './style.module.scss'

type Props = {
  storybook?: boolean
}

export const LoadingIntroduction: React.VFC<Props> = ({ storybook = false }): JSX.Element => {
  const { loading, myComponent } = useLoadingIntroduction(storybook)
  const router = useRouter()

  return (
    <div
      ref={myComponent}
      className={`${styles.wrapper} loading-introduction ${loading ? `is-active` : ''} ${storybook ? 'is-storybook' : ''}`}
      style={{ backgroundImage: `${router ? `url("${router.basePath}/webps/shutter.webp")` : 'url("webps/shutter.webp")'}` }}
    >
      Loading
      <span className={`${styles.dot} is-first`}>.</span>
      <span className={`${styles.dot} is-second`}>.</span>
      <span className={`${styles.dot} is-third`}>.</span>
    </div>
  )
}
