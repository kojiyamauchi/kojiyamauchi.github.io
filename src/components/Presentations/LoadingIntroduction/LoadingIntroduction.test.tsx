/*
  Loading Introduction Component Unit Test.
*/

import '@testing-library/jest-dom/extend-expect'

import { render /* ,act,screen */ } from '@testing-library/react'
import { Provider } from 'react-redux'

import { store } from '@/ducks'

import { LoadingIntroduction } from './Component'
// import userEvent from '@testing-library/user-event'

describe('Loading Introduction Component Unit Test', () => {
  it('Snap Shot Testing', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <LoadingIntroduction />
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
