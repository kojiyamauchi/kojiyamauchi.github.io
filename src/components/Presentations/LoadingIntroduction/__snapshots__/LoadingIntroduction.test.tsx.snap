// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Introduction Component Unit Test Snap Shot Testing 1`] = `
<DocumentFragment>
  <div
    class="wrapper loading-introduction is-active "
    style="background-image: url(webps/shutter.webp);"
  >
    Loading
    <span
      class="dot is-first"
    >
      .
    </span>
    <span
      class="dot is-second"
    >
      .
    </span>
    <span
      class="dot is-third"
    >
      .
    </span>
  </div>
</DocumentFragment>
`;
